src/crispy/result.h

```diff
@@ -146,13 +151,13 @@ class result
 
     [[nodiscard]] constexpr const T* operator->() const noexcept { return &std::get<value_type>(_data); }
     [[nodiscard]] constexpr T* operator->() noexcept { return &std::get<value_type>(_data); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr const T& operator*() const& noexcept { return std::get<value_type>(_data); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr T& operator*() & noexcept { return std::get<value_type>(_data); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr const T&& operator*() const&& noexcept { return std::move(std::get<value_type>(_data)); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
```

Comment:
wasn't it the idea to get rid of SFINAE (std::enable_if)? Isn't it possible to add a `requires` modifier to these member functions, such that they're only selected for function resolving when the requires clause is met? 🤔

Reply:
We can add requires, but we need template instantiation mechanics for it, so here we would need to create dummy template argument